### ISSUE TITLE
Allow providing own id during project creation

### ISSUE THREAD

**Issue by gagarinfan** [auditumio/auditum#146]
Hi! 

Much appreciate your work here!
I would like to ask whether it's possible to support project creation providing own ID. Currently it's generated by the Auditum.

For example:
```
POST /projects
{
  "project": {
    "id": "12345-12345-54321",
    "display_name": "some display name",
    "update_record_enabled": true,
    "delete_record_enabled": true
  }
}
```

**Comment by hypnoglow** [comment]
Hello @gagarinfan , glad you found it useful!

Could you elaborate on your use case here?

By current design, the project id is generated internally, so we don't have to validate it from the outside to be valid for the system. I'm not ready to accept the idea right away, but we can discuss.

As an alternative, what if we add another field like `extenal_id` to the project so any custom value can be stored there? If so, what use cases do we need to support, e.g. searching records by `external_id`?

**Comment by gagarinfan** [comment]
Hi @hypnoglow, thanks for the response. 

Don't want to jump into details, but in short:  my app creates a lot of projects with 24 alpha-numeric characters, for example: `6299d91c04dd5dd5d81c99bb`, `63c7db185947aee640591696`. It provides me sort of a multi-tenancy, These projects are independen and for each of them I want to have separate Auditum project.

Having many projects complicates searching as I need to iterate thru all Auditum projects to find "my project" by `display_name`

What I would like to achieve is to be able to GET Auditum project by my own ID with no need for iterating. We can achieve that by either allowing creating Auditum projects with own id OR the thing you suggested; `external_id` filed that I pass during Auditum project creation and can use later for searching purposes. For example:

```
GET /api/v1alpha1/projects/{external_id}
```
where `external_id` is my app's project ID. That would do the job 🙏

**Comment by hypnoglow** [comment]
Thank you for clarifying, we'll se what we can do here, hopefully soon!

**Comment by gagarinfan** [comment]
Great. Thanks!

**Comment by hypnoglow** [comment]
Hi @gagarinfan 

Please take a look at proposed implementation: #151, especially API ([swagger](https://github.com/auditumio/auditum/pull/151/files#diff-644c4b5b89a87b130fe140481e89eddb4a48d0c564b83ba837a73b0ffdadc469), [proto](https://github.com/auditumio/auditum/pull/151/files#diff-f130703cdafb2d2f13e2948f40ddc6fef2cb76707cb5e11a0334330395366c7e))

In short:
- New field `external_id` in a project, the value can be provided when creating project
- [ListProjects](https://auditum.io/reference/redoc/v1alpha1#tag/Projects/operation/ListProjects) operation supports a filter `external_ids`, where you can provide one or more ids to retrieve projects. Providing one id will result either in one project or no projects, meaning it does not exist.

If that would work for you, I'll add more tests and documentation, and include this feature in the next release.

**Comment by gagarinfan** [comment]
Hey @hypnoglow !

Looks good! Thanks

**Comment by hypnoglow** [comment]
Hi @gagarinfan 

This feature is now available in the latest [v0.3.0 release](https://github.com/auditumio/auditum/releases/tag/v0.3.0).

Thank you for your contribution!
